apps/browser/qml/pages/components/PopUpMenuItem.qml

```diff
@@ -0,0 +1,215 @@
+                }
+            }
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-incognito-new"
+                //% "New private tab"
+                text: qsTrId("sailfish_browser-la-new_private_tab")
+                onClicked: {
+                    webView.privateMode = true
+                    overlay.toolBar.enterNewTabUrl()
+                }
+            }
+        }
+
+        Column {
+            width: parent.width
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-search-on-page"
+                enabled: webView.contentItem
+                //% "Search on page"
+                text: qsTrId("sailfish_browser-la-search_on_page")
+
+                onClicked: {
+                    overlay.toolBar.findInPageActive = true
+                    overlay.toolBar.findInPage()
+                }
+            }
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-add-to-grid"
+                //% "Add to apps grid"
+                text: qsTrId("sailfish_browser-la-add_to_apps_grid")
+                onClicked: pageStack.animatorPush("AddToAppGridDialog.qml",
+                                                  {
+                                                      "url": url,
+                                                      "title": title,
+                                                      "desktopBookmarkWriter": desktopBookmarkWriter,
+                                                      "bookmarkWriterParent": pageStack
+                                                  })
+            }
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-file-download-as-pdf"
+                //% "Save web page as PDF"
+                text: qsTrId("sailfish_browser-la-save_as-pdf")
+
+                onClicked: {
+                    if (DownloadManager.pdfPrinting) {
+                        pdfPrintingNotice.show()
+                    } else {
+                        overlay.toolBar.savePageAsPDF()
+                    }
+                }
+            }
+        }
+
+        TextSwitch {
+            // When changing theme or orientation, the offset changes. Strange behavior.
+            leftMargin: Theme.paddingLarge * 2 + (Theme.colorScheme === Theme.LightOnDark ? Theme.paddingMedium * 2 : Theme.paddingMedium)
```

Comment:
```suggestion
            leftMargin: (root.iconWidth / 2) + root.horizontalOffset - (Theme.itemSizeExtraSmall / 2) + Theme.paddingLarge + Theme.paddingMedium
            _label.anchors.leftMargin: (root.iconWidth / 2) - root.horizontalOffset - (Theme.itemSizeSmall / 2) + Theme.paddingMedium
```
I've tested now and this change does fix it for me. While the icon for this is still pending, I'd suggest to make this change so that the we can still merge the other changes into devel in the meantime.

Reply:
Redesigned to use an icon